src/js/edu-benefits/definitions/veteranId.js

```diff
@@ -0,0 +1,44 @@
+import _ from 'lodash/fp';
+
+import * as ssn from '../../common/schemaform/definitions/ssn';
+
+// TODO: move vaFileNumber out of 1995 and into a shared backend definition
+import fullSchema1995 from 'vets-json-schema/dist/change-of-program-schema.json';
+const {
+  vaFileNumber
+} = fullSchema1995.definitions;
+
+
+export const uiSchema = {
+  veteranSocialSecurityNumber: _.assign(ssn.uiSchema, {
+    'ui:required': (form) => !form.veteranId['view:noSSN']
```

Comment:
Most of the time we use `_.get` here for nested properties, since nested objects are sometimes undefined (it depends on how you interact with the form).

`_.get('veteranId.view:noSSN', form)` will return undefined if the veteranId object doesn't exist.